django/db/migrations/state.py

```diff
@@ -146,6 +170,8 @@ def alter_model_options(self, app_label, model_name, options, option_keys=None):
             for key in option_keys:
                 if key not in options:
                     model_state.options.pop(key, False)
+        if 'all_relations' not in self.__dict__:
+            self.resolve_fields_and_relations()
```

Comment:
Is there a reason this usage of the pattern is different from all the others?

Reply:
The `alter_model_options` will alter the options of a table. So in case we alter a `ManytoMany` field we will need to populate the registry with the defined changes. These lines of code are testes in [`test_alter_model_table_m2m`](https://github.com/django/django/blob/f8e585a5d0a308c88c9ab606e1d53a840d71520c/tests/migrations/test_operations.py#L1332).

Reply:
I would add a comment to this effect, then, so readers of the code know why it's different.